sklearn/model_selection/_search.py

```diff
@@ -525,6 +555,7 @@ def _fit(self, X, y, labels, parameter_iterable):
 
         estimator = self.estimator
         cv = check_cv(self.cv, y, classifier=is_classifier(estimator))
+        self.metric_name_ = _get_metric_name(self.scoring)
         self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
 
         n_samples = _num_samples(X)
```

Comment:
This is not your fault but there is a `len(y) != n_samples` check, that should be redundant with the `indexable` call